tf2_ros_py/tf2_ros/buffer_client.py

```diff
@@ -230,18 +230,19 @@ def unblock_by_timeout():
             while not send_goal_future.done() and not event.is_set():
                 if clock.now() > start_time + timeout + timeout_padding:
                     break
-                # TODO(vinnamkim): rclpy.Rate is not ready
-                # See https://github.com/ros2/rclpy/issues/186
-                #r = rospy.Rate(self.check_frequency)
-                sleep(1.0 / self.check_frequency)
+
+                rate.sleep()
```

Comment:
I'm not sure this code should use `rate.sleep()`. It looks like `__process_goal()` blocks on `event.wait()` below. If someone calls [`lookup_transform()`](https://github.com/ros2/geometry2/blob/f619fed7df8c0d07ea78442aa52f0cfbcee03fc7/tf2_ros_py/tf2_ros/buffer_client.py#L122) from the callback of a single threaded executor then `event.wait()` in `__process_goal()` will block that thread, meaning the executor can't call the timer callback inside of `rate.sleep()`. I think this would block forever.

Reply:
@sloretz This was documented in the PR, as well as in the method `sleep`. I should have tested for this, and running some of the examples would have made this apparent.

I dont see a way to use the `Rate` object in this case. Is the best course of action to just remove the TODO and leave it in the original state? or simply close this PR all together.

This is likely the same for the TODO in `buffer.py`. Outside the executor situation, node is an optional in the `Buffer` class as well.

https://github.com/ros2/geometry2/blob/3cc8bbe1a67790b149b09fa7a04495b1262c168a/tf2_ros_py/tf2_ros/buffer.py#L227

Reply:
> That said could the visibility of this dynamic be improved?

Yeah. It could be improved. There are a lot of places this comes up when working with the single threaded executor. Usually it can be nicely solved with coroutines, but the tf2 Python API needs a lot of work to get there.

> I dont see a way to use the Rate object in this case. Is the best course of action to just remove the TODO and leave the rest in the original state?  or simply close this PR all together.

I would recommend getting rid of the TODO and opening an issue to provide a way to timeout waiting for transforms that doesn't block the executor - like a callback when transforms are ready or something that can be `await`'d.

Reply:
Understood. @clalancette Can you approve, or advise how we should proceed? Thank you.

Reply:
If I'm understanding properly, what we have to do here is put the previous code back.  What I'll suggest is that we fix the TODO to say something like:

```
# TODO(surfertas): We can't use Rate here because it would never expire with a single-threaded executor.  See <issue-number> for ideas on how to timeout waiting for transforms that don't block the executor.
```

We'll also have to make that new issue.

Reply:
@clalancette Ive made the changes and put the previous code back. I have also opened a new issue #327 .
I will bounce some ideas off sloretz there. Thanks a lot.